src/ray/util/filesystem.h

```diff
@@ -59,4 +59,9 @@ std::string JoinPaths(std::string base, const Paths &...components) {
   (join(base, std::string_view(components)), ...);
   return base;
 }
+
+// Read the whole content for the given [fname], and return as string.
+// If any error happens, throw exception.
```

Comment:
where is exception thrown?

Reply:
fixed, glog uses exception for CHECK, while RAY_CHECK exits the program.